azuredevops/internal/service/approvalsandchecks/resource_check_approval.go

```diff
@@ -0,0 +1,129 @@
+package approvalsandchecks
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/utils/pipelineschecksextras"
+)
+
+// ResourceCheckApproval schema and implementation for branch check resources
+func ResourceCheckApproval() *schema.Resource {
+	r := genBaseCheckResource(flattenCheckApproval, expandCheckApproval)
+
+	r.Schema["approvers"] = &schema.Schema{
+		Type:     schema.TypeList,
+		MinItems: 1,
+		Required: true,
+		Elem: &schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validation.IsUUID,
+		},
+	}
+
+	r.Schema["instructions"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Optional: true,
+	}
+
+	r.Schema["requester_can_approve"] = &schema.Schema{
+		Type:     schema.TypeBool,
+		Optional: true,
+		Default:  false,
+	}
+
+	r.Schema["minimum_required_approvers"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Optional: true,
+	}
+
+	r.Schema["timeout"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Optional: true,
+		Computed: true,
+	}
+
+	return r
+}
+
+func flattenCheckApproval(d *schema.ResourceData, check *pipelineschecksextras.CheckConfiguration, projectID string) error {
+	err := doBaseFlattening(d, check, projectID)
+	if err != nil {
+		return err
+	}
+
+	if check.Settings == nil {
+		return fmt.Errorf("settings nil")
+	}
+
+	check.Type = approvalAndCheckType.Approval
+
+	settings := check.Settings.(map[string]interface{})
+
+	if instructions, found := settings["instructions"]; found {
+		d.Set("instructions", instructions)
+	} else {
```

Comment:
no need to raise an error if `instructions` not found as it is an optional property.